railties/lib/rails/application/finisher.rb

```diff
@@ -138,6 +138,9 @@ def self.complete(_state)
           # some sort of reloaders dependency support, to be added.
           require_unload_lock!
           reloader.execute
+          if app.config.eager_load
+            app.routes.eager_load!
```

Comment:
Maybe we should move the `eager_load!` call to the reloader like this:

```
def eager_load!
  execute
  route_sets.each(&:finalize!)
end
```

And here

```
if app.config.eager_load
  reloader.eager_load!
else
  reloader.execute
end
```